docs: clarify N_JOBS_PER_WORKER governs run concurrency, not request concurrency (#5042)
langchain-ai/docs: src/langsmith/agent-server-scale.mdx

For [Cloud](/langsmith/cloud-platform-features#scaling), the platform autoscales automatically and the Helm configurations below do not apply.

## Request vs. run concurrency

Two independent kinds of concurrency determine how the Agent Server scales, and they are controlled separately:

> - **Request concurrency** is how many API requests (creating runs, reading thread state, streaming results) the deployment serves at once. API servers handle requests asynchronously and scale horizontally with the number of API server replicas. Request-serving capacity is not capped by [`N_JOBS_PER_WORKER`](/langsmith/env-var-self-hosted).

Reviewer: nit: Maybe just leave N_JOBS_PER_WORKER off here? Makes sense in the context of this PR, but I think to new eyes, its cleaner to just focus on what controls this instead of what doesn't.
```suggestion
- **Request concurrency** is how many API requests (creating runs, reading thread state, streaming results) the deployment serves at once. API servers handle requests asynchronously and the request concurrency scales horizontally with the number of API server replicas.
```

Author: Done. Dropped the `N_JOBS_PER_WORKER` clause from this bullet and applied your suggested wording. The point that it does not cap request concurrency still lives in the closing paragraph, so nothing is lost.

> - **Run concurrency** is how many runs execute at once. A single queue worker executes up to `N_JOBS_PER_WORKER` runs concurrently (default 10), and total run capacity scales with the number of queue workers multiplied by `N_JOBS_PER_WORKER`.

Reviewer: I think we can make this more explicit because this is a hard cap.
```suggestion
- **Run concurrency** is how many runs execute at once. A single queue worker executes up to `N_JOBS_PER_WORKER` runs concurrently (default 10). Run concurrency is capped at the number of queue workers multiplied by `N_JOBS_PER_WORKER`.
```

Reviewer: Maybe we can link out to the N_JOBS_PER_WORKER env var here too?

Author: Done. Switched to hard-cap wording (Run concurrency is capped at the number of queue workers multiplied by `N_JOBS_PER_WORKER`) and moved the `N_JOBS_PER_WORKER` env var link onto this bullet, which is now its first mention.

> Creating a run is a fast write request: the API server persists a pending run and returns immediately, without waiting for the run to execute. If every run slot is busy, additional runs wait in the queue until a slot frees. Raising `N_JOBS_PER_WORKER` or adding queue workers increases run throughput; it does not change how many requests the deployment can serve concurrently.

Reviewer: Do we think there's a gap in the docs around the queuing section explaining this? Maybe good to link out to the architecture there.

Author: Linked `queue` out to the Run execution lifecycle section on the Agent Server page, which documents enqueue, pending, pickup, and slot release. Kept this to a link rather than expanding the queue docs for now.
